### PR TITLE
bugfix workflowstate encoding, missing generation field

### PR DESCRIPTION
**Pull request by famarting**
# Description

This PR updates the functions `EncodeWorkflowState` and `DecodeWorkflowState` so they dont use an embedded struct definition and instead use a well defined struct, additionally this PR adds the field `Generation` to the encoded data and a test to verify encoding and decoding works as expected.

## Issue reference

Please reference the issue this PR will close: #_[issue number]_

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [ ] Code compiles correctly
* [ ] Created/updated tests
* [ ] Unit tests passing
* [ ] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

File: pkg/runtime/wfengine/backends/actors/workflowstate_test.go [hunk 1]
```diff
@@ -0,0 +1,47 @@
+package actors
```
Comment (JoshVanL) [inline pkg/runtime/wfengine/backends/actors/workflowstate_test.go:1]: Add copyright header

**Review by JoshVanL — CHANGES_REQUESTED** [review]

**Review by ItalyPaleAle — DISMISSED** [review]

**Comment by yaron2** [comment]
Linter @famarting

**Comment by famarting** [comment]
tests need retrigger, these failures seem unrelated to my changes

**Review by yaron2 — APPROVED** [review]

**Review by elena-kolevska — APPROVED** [review]

**Comment by famarting** [comment]
ping, we need this 😄

**Pull request merged**